packages/pwa-kit-extension-sdk/src/react/utils/helpers.ts

```diff
@@ -23,3 +23,40 @@ export const applyHOCs = <T extends React.ComponentType<any>>(Component: T, hocs
         return hoistNonReactStatics(WrappedComponent, AccumulatedComponent) as T
     }, Component)
 }
+
+/**
+ * Applies a cache to a method on an instance. The cache is stored in a property on the instance.
+ *
+ * @param instance - The instance on which to apply the cache.
+ * @param methodName - The name of the method to cache.
+ * @param cacheProperty - The name of the property to use for caching.
+ */
+export function cacheMethodResult(instance: any, methodName: string, cacheProperty: string) {
+    const originalMethod = instance[methodName]
+
+    if (typeof originalMethod === 'function') {
+        instance[methodName] = function (...args: any[]) {
+            if (instance[cacheProperty] !== undefined && instance[cacheProperty] !== null) {
+                return instance[cacheProperty]
+            }
+
+            const result = originalMethod.apply(instance, args)
+
+            if (result instanceof Promise) {
+                return (instance[cacheProperty] = result.then((resolved) => {
+                    instance[cacheProperty] = resolved
+                    return resolved
+                }))
+            }
+
+            return (instance[cacheProperty] = result)
+        }
+    }
+}
+
+/**
+ * Checks if the code is running on the server side.
+ *
+ * @returns {boolean} `true` if running on the server, `false` if running on the client.
+ */
+export const isServerSide = () => typeof window === 'undefined'
```

Comment:
I chose to make this a util function to make testing easier, but wasn't sure if it should be in `pwa-kit-react-sdk`